internal/ignition/ignition_test.go

```diff
@@ -607,3 +610,97 @@ var _ = Describe("Openshift cluster ID extraction", func() {
 		Expect(err.Error()).To(Equal("no ClusterID field in cvo-overrides file"))
 	})
 })
+
+var _ = Describe("Generator UploadToS3", func() {
+	var (
+		ctx          = context.Background()
+		ctrl         *gomock.Controller
+		mockS3Client *s3wrapper.MockAPI
+	)
+
+	generator := installerGenerator{
+		log:     log,
+		workDir: workDir,
+	}
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		mockS3Client = s3wrapper.NewMockAPI(ctrl)
+
+		generator.s3Client = mockS3Client
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	mockUploadFile := func() *gomock.Call {
+		return mockS3Client.EXPECT().UploadFile(gomock.Any(), gomock.Any(), gomock.Any())
+	}
+
+	mockUploadObjectTimestamp := func() *gomock.Call {
+		return mockS3Client.EXPECT().UpdateObjectTimestamp(gomock.Any(), gomock.Any())
+	}
+
+	It("disabled host", func() {
+		cluster.Hosts = []*models.Host{
+			{Status: swag.String(models.HostStatusDisabled)},
+		}
+		generator.cluster = cluster
+
+		mockUploadFile().Return(nil).Times(len(fileNames))
+		mockUploadObjectTimestamp().Return(true, nil).Times(len(fileNames))
+
+		Expect(generator.UploadToS3(ctx)).ShouldNot(HaveOccurred())
```

Comment:
This applies to all of these expectations, and is just a style nit, but this reads strangely to me.
The way it's written makes it feel like an expectation that the method should not have been called rather than an assertion about the error response.

I think one of these would be better:

This makes it clear that the assertion is on a value rather than some event that we have to infer
```go
Expect(generator.UploadToS3(ctx)).To(BeNil())
```

or

This reads a bit more like English and also makes it clear that we're talking about an error
```go
err := generator.UploadToS3(ctx)
Expect(err).NotTo(HaveOccurred())
```

Reply:
then in case of error it should be `Expect(generator.UploadToS3(ctx)).NotTo(BeNil())`?